File: class2relational/src/main/java/org/atlanmod/model/scratch/classModel/ClassLink.java
package org.atlanmod.model.scratch.classModel;

import org.atlanmod.Link;

import java.util.List;

public class ClassLink implements Link<ClassElement, String, List<ClassElement>> {

    private ClassElement source;
    private String ref;
    private List<ClassElement> target;

    public ClassLink(ClassElement source, String ref, List<ClassElement> target){
        this.source = source;
        this.ref = ref;
        this.target = target;
    }

    @Override
    public ClassElement getSource() {
        return this.source;
    }

    @Override
    public String getReference() {
        return this.ref;
    }

    @Override
    public List<ClassElement> getTarget() {
        return this.target;
    }
}

File: sparktl-transformations/src/main/scala/org/atlanmod/class2relational/model/classmodel/ClassLink.scala
package org.atlanmod.class2relational.model.classmodel

import org.atlanmod.tl.model.impl.dynamic.DynamicLink
import org.atlanmod.tl.util.ListUtils

abstract class ClassLink(type_ : String, source: ClassElement, target: List[ClassElement])
  extends DynamicLink(type_, source, target){

    override def equals(o: Any): Boolean = {
        o match {
            case obj: ClassLink => type_.equals(obj.getType) & source.equals(obj.getSource) &
              ListUtils.eqList(obj.getTarget, target)
            case _ => false
        }
    }

    override def weak_equals(o: Any): Boolean = {
        o match {
            case obj: ClassLink =>
                source.weak_equals(obj.getSource) &&
                  type_.equals(obj.getType) &&
                  ListUtils.weak_eqList(obj.getTarget, target)
            case _ => false
        }
    }



}
